apps/ejabberd/src/mongoose_api_users.erl

```diff
@@ -98,7 +98,7 @@ maybe_register_user(Username, Host, Password) ->
             ?ERROR;
         {error, exists} ->
             maybe_change_password(Username, Host, Password);
-        ok ->
+        _ ->
```

Comment:
Me too, but currently there is no way I know of to match on mongoose_stanza - it happens to be a map now, but it may change in the future. And this brings us back to Michał's idea that maybe :run_fold should return {ok, Acc} instead of just Acc, which would solve this type of problems.